shared/process-gecko-analysis.sh

```diff
@@ -51,5 +44,7 @@ sort analysis-files.list | uniq | parallel --halt now,fail=1 "RUST_LOG=info $MOZ
 
 date
 
-# Free up disk space, we don't need these per-platform analysis files any more.
-rm -rf analysis-*
+# We previously would delete the generated-* and analysis-* files and
+# directories at this point because they were no longer necessary for subsequent
+# processing, but this has greatly complicated diagnosing analysis problems,
+# so we now leave them around.
```

Comment:
There's a chance this will cause the mozilla-releases indexer to run out of disk space. Currently the index volume on that channel is at 72% capacity, and it has three repositories (-beta, -release, -esr68) that will run this processing and leave these files lying around. Might want to check that it doesn't blow up. But note also that we keep the downloaded tarballs, so in case of diagnosing analysis errors, the dev doing the debugging could just re-extract the files from there. Not sure how much of a hassle that would be.

Reply:
Keeping the tarballs seems like a good compromise[1] and great idea, thank you.  I hadn't run the disk usage on the leftovers... there's a LOT of leftovers (~5G) per dir... I was thinking they got emptied out more than they do.

1: And arguably better than keeping the directories around since I'm mainly interested in the pre-merge state, not the post-merge state.  And that way they can be extracted from the S3 mount to the local instance mount directly, which is a win over trying to move everything back into place.

Reply:
Okay, the footer block will now be:
```bash
# Delete the generated-* and analysis-* directories, but retain the tarballs
# for ease of investigation.  (The tarballs are still available from taskcluster
# though, and fetch-tc-artifcats.sh knows how to re-fetch them, so if we don't
# want to waste the space, we could just make it easier to re-fetch them so
# there's a one-liner in mozsearch's `docs/aws.md`.)
rm -rf generated-*/
rm -rf analysis-*/
```

The net change is we leave the tarballs around.  This does represent a deferral of the removal of the `analysis-*/__GENERATED__` sub-trees to this block.  I had eliminated that removal when I had been (tired and) confused and hadn't realized that rust-indexer wasn't generating the nsIChannel.rs analysis output (because I didn't have RUST_LOG=info in process-tc-artifacts.sh), rather than it having been deleted.  This was the objdir/generated (files) confusio that led to the late rename.

Technically that `analysis-*/__GENERATED__` removal could go back, but since the analysis-dirs.list and analysis-files.list explicitly exclude the __GENERATED__ sub-paths and we don't need the disk space at that instance, there's also no harm in the deferral.  So I'm going to leave it as-is, but I don't have strong opinions other than not wanting to risk somehow breaking things.  (The main difference seems like the earlier deletion would resut in a different disk placement for the non-generated analysis merges, but we're not dealing with spinning disks anymore, so disk placement is largely immaterial.)

Reply:
Sounds good to me!